config/config_test.go

```diff
@@ -2300,3 +2301,52 @@ func TestScrapeConfigDisableCompression(t *testing.T) {
 
 	require.False(t, got.ScrapeConfigs[0].EnableCompression)
 }
+
+func TestScrapeConfigNameValidationSettings(t *testing.T) {
+	model.NameValidationScheme = model.UTF8Validation
+	defer func() {
+		model.NameValidationScheme = model.LegacyValidation
+	}()
```

Comment:
Let's add this in 3.0 branch I would say. For main branch lets keep it default to legacy as it should?

Reply:
no, it needs to enable the default validation in model or else there will hit the error here: https://github.com/prometheus/prometheus/blob/main/config/config.go#L778-L780